streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -510,4 +526,17 @@ private void initPartitionGrouper(StreamsConfig config, StreamThread thread) {
 
         partitionAssignor.onAssignment(assignments.get("client"));
     }
+
+    public static class StateListenerStub implements StreamThread.StateListener {
+        public int numChanges = 0;
+        public StreamThread.State oldState;
+        public StreamThread.State newState;
+
+        @Override
+        public void onChange(final StreamThread.State newState, final StreamThread.State oldState) {
+            this.numChanges++;
```

Comment:
Maybe also check that the current `newState` is equal to the passed in `oldState`?